hack/update-mocks.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates mock files using mockgen.
+# Usage: `hack/update-mocks.sh`.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KARMADA_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# Explicitly opt into go modules, even though we're inside a GOPATH directory
+export GO111MODULE=on
+export PATH=$PATH:$(go env GOPATH)/bin
+
+trap EXIT
+
+echo 'installing mockgen'
+source "${KARMADA_ROOT}"/hack/util.sh
+echo -n "Preparing: 'mockgen' existence check - "
+if [ ! $(util::cmd_exist mockgen) ]; then
+  util::install_tools "github.com/golang/mock/mockgen" "latest"
```

Comment:
We can't install `latest`, the generated file may change over `mockgen` versions.
```suggestion
  GO111MODULE=on go install "github.com/golang/mock/mockgen"
```
This way installing `mockgen` with the version in the vendor. For CI we need a consistent version.

Reply:
+1